tracing-core/src/metadata.rs

```diff
@@ -122,7 +123,7 @@ impl<'a> Metadata<'a> {
     /// Construct new metadata for a span or event, with a name, target, level, field
     /// names, and optional source code location.
     pub const fn new(
```

Comment:
while with this we now have general support for dynamic creation, the current API, still won't allow for that. Probably want to add another `fn dynamic_new` where we can pass `Cow<'a, str>` into for dynamic dispatching.

Reply:
I am afraid so ....